text/0000-api-jwt-authentication.md

```diff
@@ -0,0 +1,322 @@
+---
+layout: default
+title: JWT Authentication for Fabric APIs
+nav_order: 3
+---
+
+- Feature Name: api_jwt_authentication
+- Start Date: 2021-01-11
+- RFC PR:
+- Fabric Component: orderer, peer
+- Fabric Issue:
+
+# Summary
+
+This document proposes an OAuth2 compatible authentication mechanism for MSP
+identities that produces authorization tokens suitable for use with APIs
+exposed by Fabric.
```

Comment:
@sykesm  Dear Matt, you mentioned about channel participation API in motivation and in later section I see `The authentication mechanism described here can also be applied to gRPC services to support activities like chaincode install and query operations.`  If this RFC aims to provide one-off solution to provide ALL mutual TLS APIs with another authentication mechanism, that is more welcomed. Otherwise web developers will face a more optional but complex situation that “Sometime we A, sometime we B, since Fabric is transforming but on the halfway”

Reply:
The target for this work is the channel participation API. It is the only REST (not gRPC) API that we have that influences data processing in the network.

Reply:
So this work target will not including roadmap about gRPC endpoint auth. Is it？

Reply:
This RFC is explicitly targeted at the channel participation API and I'd like it to remain focused on that. A narrow scope keeps it containable and simple.

While the same technology can be used to authenticate with gRPC services, that is not in scope.

>  aims to provide one-off solution
I also want to say that I disagree with this characterization. We have gone out of our way to build on and around standard protocols to avoid this becoming a "one-off" solution.